### PR TITLE
add shoulda-callback-matchers to gemfile, test callback in category m…

### PR DESCRIPTION
**Pull request by cmacaulay**
…odel test
- Commented out start of test for checking out

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -53,6 +53,7 @@ end
 
 group :test do
   gem 'database_cleaner'
+  gem 'shoulda-callback-matchers', '~> 1.1.1'
```
Comment (cmacaulay) [inline Gemfile:56]: Here's the new gem.

File: app/models/category.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 class Category < ApplicationRecord
-
   validates :name, presence: true, uniqueness: true
```
Comment (cmacaulay) [inline app/models/category.rb:2]: Deleted a line.

File: spec/models/category_spec.rb [hunk 3]
```diff
@@ -1,9 +1,16 @@
-require 'rails_helper'
+require "rails_helper"
```
Comment (cmacaulay) [inline spec/models/category_spec.rb:1]: Changed all quotations to double quotes.

File: spec/models/category_spec.rb [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
```
Comment (cmacaulay) [inline spec/models/category_spec.rb:2]: Added new contexts, before all were lumped into validations.

File: spec/models/category_spec.rb [hunk 5]
```diff
@@ -1,9 +1,16 @@
...
     it { is_expected.to validate_presence_of(:name) }
     it { is_expected.to validate_uniqueness_of(:name) }
+  end
+
+  context "relationships" do
     it { should have_many(:items) }
   end
+
+  context "callbacks" do
+    it { is_expected.to callback(:assign_slug).before(:save) }
```
Comment (cmacaulay) [inline spec/models/category_spec.rb:14]: Here's the new callback test.

**Review by cmacaulay — COMMENTED** [review]
All tests passing.

File: spec/models/category_spec.rb (hunk 5 above)
Comment (meyerhoferc) [reply spec/models/category_spec.rb:14]: 🔥

File: Gemfile (hunk 1 above)
Comment (meyerhoferc) [reply Gemfile:56]: awesome!

**Comment by meyerhoferc** [comment]
I approve! Ready for review & merge @valerietrudell @CjMoore

File: spec/models/category_spec.rb (hunk 5 above)
Comment (valerietrudell) [inline spec/models/category_spec.rb]: looks good

**Pull request merged**